homeassistant/components/eufylife_ble/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The EufyLife integration."""
+from __future__ import annotations
+
+from eufylife_ble_client import EufyLifeBLEDevice
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.match import ADDRESS, BluetoothCallbackMatcher
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant, callback
+
+from .const import CONF_MODEL, DOMAIN
+from .models import EufyLifeData
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up EufyLife device from a config entry."""
+    address = entry.unique_id
+    assert address is not None
+
+    model = entry.data[CONF_MODEL]
+    client = EufyLifeBLEDevice(model=model)
+
+    @callback
+    def _async_update_ble(
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Update from a ble callback."""
+        client.set_ble_device_and_advertisement_data(
+            service_info.device, service_info.advertisement
+        )
+        if not client.advertisement_data_contains_state:
+            hass.async_create_task(client.connect())
```

Comment:
It would be nicer if the lib took care of connecting only if it wasn't already connected and only created a task if it needed to.  Without digging into the lib, it looks like it might create multiple connect tasks here

Reply:
This is already handled in the library. I borrowed the logic from your `led-ble` library, which was very helpful as an example.